COVID_19_model.ipynb

```diff
@@ -1652,8 +1674,140 @@
         "    )\n",
-        "    countries_plot_data[studied_country] = country_plot_data\n",
-        "    \n",
+        "    countries_plot_data[studied_country] = country_plot_data"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "c8A48h0SKdlq",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "#@title Single plotly combined plot of the best guess forecasts\n",
+        "for cumulative in [False, True]:\n",
+        "  plot_countries = countries_plot_data.keys()\n",
+        "  num_plot_countries = len(plot_countries)\n",
+        "  fig = go.Figure()\n",
+        "\n",
+        "  for i, country in enumerate(plot_countries):\n",
+        "    print(\"Adding traces for country {} of {}\".format(i+1, num_plot_countries))\n",
+        "    country_plot_data = countries_plot_data[country]\n",
+        "    simple_model_data = analysis_df.iloc[(\n",
+        "      analysis_df[[country_col_name]].values[:, 0] == country)]\n",
+        "\n",
+        "    # pdb.set_trace()\n",
+        "    traces = plot_country_posterior(\n",
+        "      country,\n",
+        "      country_plot_data.posterior_samples,\n",
+        "      simple_model_data,\n",
+        "      save_plot=False,\n",
+        "      pred_vals=country_plot_data.pred_vals,\n",
+        "      cumulative=cumulative,\n",
+        "      show_plot=False,\n",
+        "    )\n",
+        "\n",
+        "    # Add all traces to the correct subplot\n",
+        "    marker_color = get_color(i)\n",
+        "    for j, t in enumerate(traces):\n",
+        "      name = country\n",
+        "      if j == len(traces)-2:\n",
+        "        name += \" Forecast\"\n",
+        "        line_dash = \"dot\"\n",
+        "      elif j == len(traces)-1:\n",
+        "        name += \" Actual\"\n",
+        "        line_dash = None\n",
+        "      else:\n",
+        "        continue\n",
+        "      t.legendgroup = name\n",
+        "      t.name = name\n",
+        "      t.showlegend = True\n",
+        "      t.marker.color = marker_color\n",
+        "      t.line.dash = line_dash\n",
+        "      fig.add_trace(t)\n",
+        "\n",
+        "  title_text = \"Cumulative deaths forecast\" if cumulative else (\n",
+        "      \"Daily deaths foreast\")\n",
+        "  fig.update_layout(title_text=title_text, title_x=0.5, font=dict(size=16))\n",
+        "\n",
+        "  fig.update_layout(xaxis=go.layout.XAxis(title=go.layout.xaxis.Title(\n",
+        "    font=dict(size=axis_label_font_size))),\n",
+        "                  yaxis=go.layout.YAxis(title=go.layout.yaxis.Title(\n",
+        "                      text=\"Death count\",\n",
+        "                      font=dict(size=axis_label_font_size))),\n",
+        "                  font=dict(size=tick_font_size))\n",
+        "  # fig.update_yaxes(range=[0, 1000])\n",
```

Comment:
unused

Reply:
Is this comment just about the yrange?
I left it in since we might want to reuse that at some point but am happy to delete

Reply:
Not a big deal then